### PR TITLE
Accessibility fixes: Navigation

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done

Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/3225

## QA

- Pull code
- Run `./run`
- review code changes in all navigation examples

### CODE REVIEW

File: templates/docs/examples/templates/z-index.html [hunk 1]
```diff
@@ -64,8 +64,8 @@
           </ul>
         </li>
       </ul>
-      <ul class="p-navigation__items" role="menu">
-        <li class="p-navigation__item p-subnav" role="menuitem" id="link-1">
+      <ul class="p-navigation__items">
+        <li class="p-navigation__item p-subnav" id="link-2">
```
Comment (bartaz) [inline templates/docs/examples/templates/z-index.html:68]: There already is a `link-2` above.

```suggestion
        <li class="p-navigation__item p-subnav" id="link-4">
```

File: templates/docs/examples/patterns/navigation/subnav.html [hunk 2]
```diff
@@ -69,8 +69,8 @@
           </ul>
         </li>
       </ul>
-      <ul class="p-navigation__items" role="menu">
-        <li class="p-navigation__item p-subnav" role="menuitem" id="link-1">
+      <ul class="p-navigation__items">
+        <li class="p-navigation__item p-subnav" id="link-2">
```
Comment (bartaz) [inline templates/docs/examples/patterns/navigation/subnav.html:73]:
```suggestion
        <li class="p-navigation__item p-subnav" id="link-4">
```

File: templates/docs/examples/patterns/navigation/subnav-dark.html [hunk 3]
```diff
@@ -69,8 +69,8 @@
           </ul>
         </li>
       </ul>
-      <ul class="p-navigation__items" role="menu">
-        <li class="p-navigation__item p-subnav" role="menuitem" id="link-1">
+      <ul class="p-navigation__items">
+        <li class="p-navigation__item p-subnav" id="link-2">
```
Comment (bartaz) [inline templates/docs/examples/patterns/navigation/subnav-dark.html:73]:
```suggestion
        <li class="p-navigation__item p-subnav" id="link-4">
```

**Comment by lyubomir-popov** [comment]
I've applied your suggestions, changing label back to "review needed"

**Review by bartaz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**